File: shared/src/commonTest/kotlin/com/mbta/tid/mbta_app/mocks/MockPinnedRoutesRepository.kt
package com.mbta.tid.mbta_app.mocks

import com.mbta.tid.mbta_app.repositories.IPinnedRoutesRepository

class MockPinnedRoutesRepository : IPinnedRoutesRepository {
    private var pinnedRoutes: Set<String> = emptySet()

    override suspend fun getPinnedRoutes(): Set<String> {
        return pinnedRoutes
    }

    override suspend fun setPinnedRoutes(routes: Set<String>) {
        pinnedRoutes = routes
    }
}

File: iosApp/iosAppTests/Mocks/MockPinnedRoutesRepository.swift
//
//  MockPinnedRoutesRepository.swift
//  iosAppTests
//
//  Created by Brandon Rodriguez on 4/17/24.
//  Copyright © 2024 MBTA. All rights reserved.
//

import Shared

class MockPinnedRoutesRepository: IPinnedRoutesRepository {
    var pinnedRoutes: Set<String> = []

    func __getPinnedRoutes() async throws -> Set<String> {
        pinnedRoutes
    }

    func __setPinnedRoutes(routes: Set<String>) async throws {
        pinnedRoutes = routes
    }
}
